packages/matter-node.js/src/matter/common/DeviceTypes.ts

```diff
@@ -3,49 +3,221 @@
  */
+
+import * as MatterClusters from "@project-chip/matter.js"; // TODO: Adjust after code move
 import { MatterDeviceLibrarySpecificationV1_0 } from "@project-chip/matter.js";
 
-export const DEVICE = {
+export enum DeviceClasses {
+    /** Node device type. */
+    Node = "Node",
+
+    /**
+     * Application device type.
+     * Application devices types are typically the most common endpoints on a node and in the network.
+     */
+    App = "App",
+
+    /**
+     * Simple device type.
+     * A Simple device type supports local control that is persistent, independent, and unsupervised.
+     */
+    Simple = "Simple",
+
+    /**
+     * Dynamic device type.
+     * A Dynamic device type supports intelligent and supervisory services, such as commissioning,
+     * monitoring, trend analysis, scheduling and central management. A dynamic device type is an
+     * application device type.
+     */
+    Dynamic = "Dynamic",
+
+    /** There exists a client application cluster on the endpoint. */
+    Client = "Client",
+
+    /** There exists a server application cluster on the endpoint. */
+    Server = "Server",
+
+    /** The device type is composed of 2 or more device types. */
+    Composed = "Composed",
+
+    /** Composed device type that is composed of 2 or more endpoints with the same device type. */
+    Multiple = "Multiple",
+
+    /** The endpoint is an Initiator for Zigbee EZ-Mode Finding & Binding. */
+    "EZInitiator" = "EZ-Initiator",
+
+    /** The endpoint is a Target for Zigbee EZ-Mode Finding & Binding. */
+    "EZTarget" = "EZ-Target",
+
+    /**
+     * The endpoint represents a Bridged Device, for which information about the state of
+     * its power source is available to the Bridge
+     */
+    BridgedPowerSourceInfo = "BridgedPowerSourceInfo",
+}
+
+export interface DeviceTypeDefinition {
+    name: string;
+    code: number;
+    class: DeviceClasses;
+    superSet?: string;
+    revision: number;
+    requiredServerClusters: number[];
+    optionalServerClusters: number[];
+    requiredClientClusters: number[];
+    optionalClientClusters: number[];
+}
+
+export const DEVICE: { [key: string]: DeviceTypeDefinition } = {
     // Utility Device Types
+    // A Utility device type supports configuration and settings.
     /**
      * This represents a Root Node for devices.
      * @see {@link MatterDeviceLibrarySpecificationV1_0} § 2.1
      */
-    ROOT: { name: "MA-rootdevice", code: 0x0016 },
+    ROOT: {
+        name: "MA-rootdevice",
+        code: 0x0016,
+        class: DeviceClasses.Node,
+        revision: 1,
+        requiredServerClusters: [
+            MatterClusters.BasicInformationCluster.id,
+            MatterClusters.AccessControlCluster.id,
+            MatterClusters.GroupKeyManagementCluster.id,
+            MatterClusters.GeneralCommissioningCluster.id,
+            MatterClusters.AdminCommissioningCluster.id,
+            MatterClusters.OperationalCredentialsCluster.id,
```

Comment:
Standard calls this "Node Operational Credentials"

Reply:
Yes but we have that other name relatively long. I might rename it to other name whendoing big code move. I add a todo for now!